assets/src/edit-story/components/library/panes/text/textPane.js

```diff
@@ -32,17 +33,67 @@ import { Section, MainButton, SearchInput } from '../../common';
 import { FontPreview } from '../../text';
 import useLibrary from '../../useLibrary';
 import { Pane } from '../shared';
+import { dataFontEm } from '../../../../units';
+import { useHistory } from '../../../../app/history';
 import paneId from './paneId';
 import { PRESETS, DEFAULT_PRESET } from './textPresets';
 
 const SectionContent = styled.p``;
 
+// The format is:
+// { presetToBeAddedName: [list of previous preset names that need to be considered]
+// If the previous preset is not in the list of names, no special rules.
+// If the preset being added is not in the rules at all -- no special rules.
+// @todo Text staggering to be done as part of https://github.com/google/web-stories-wp/issues/1206
+const POSITIONING_RULES = {
+  'heading-2': ['heading-1'],
+  'heading-3': ['heading-1', 'heading-2'],
+  paragraph: ['heading-1', 'heading-2', 'heading-3'],
+};
+
+const POSITION_MARGIN = dataFontEm(1);
+
 function TextPane(props) {
   const { insertElement } = useLibrary((state) => ({
     insertElement: state.actions.insertElement,
   }));
+  const {
+    state: { versionNumber },
+  } = useHistory();
+
   const { showTextSets, showTextAndShapesSearchInput } = useFeatures();
 
+  const lastPreset = useRef(null);
+
+  const getTopPosition = (name, element) => {
+    const { y } = element;
+    // If we don't have any special rules, just add the default.
+    if (
+      !POSITIONING_RULES[name] ||
+      !lastPreset.current ||
+      !POSITIONING_RULES[name].includes(lastPreset.current.name)
+    ) {
+      return y;
+    }
+    // If the difference between the new and the previous version number is not 1,
+    // the preset wasn't clicked right after the previous.
+    if (versionNumber - lastPreset.current.versionNumber !== 1) {
+      lastPreset.current = null;
+      return y;
+    }
+    // Use the positioning of the previous preset + add the previous preset's
+    // approximate height + lineHeight if more than 1 + position margin.
```

Comment:
What if the previous element had more than one line? Maybe in a translation scenario `"Heading 1"` will break to multiple lines? Or is that not a worry? Could we not just use the actual height (in data pixels) of the element added based on `element.height`?

Reply:
Hmm, it's very unlikely that translating "Heading 1" would fill up the whole line. If that's the case then the vertical centering currently done for the default text position is also not correct (for this specific translation case).

For positioning the presets we could use the inserted elements actual height though -- should be more reliable anyway. Will change that.